src/main/java/uk/ac/ebi/ega/egacryptor/service/TaskExecutorService.java

```diff
@@ -74,8 +90,11 @@ public void execute(final List<FileToProcess> fileToProcessList, final int noOfT
         for (final Future<String> stringFuture : futureList) {
             try {
                 stringFuture.get();
-            } catch (InterruptedException | ExecutionException e) {
-                LOGGER.error("Error while iterating over future list - " + e.getMessage(), e);
+            } catch (ExecutionException e) {
+                LOGGER.error("Error while iterating over future list: {}", e.getMessage(), e);
+            } catch (InterruptedException e) {
+                LOGGER.error("Error while iterating over future list: {}", e.getMessage(), e);
+                Thread.currentThread().interrupt();
```

Comment:
Sonar told me. I have ~no~ only a vague idea what this does. Please correct this line if I did something silly.